File: leaderboards/app/modules/LeaderboardsModule.java
package modules;

import com.google.inject.AbstractModule;
import interfaces.*;
import implementors.*;

public class LeaderboardsModule extends AbstractModule {
    @Override
    protected void configure() {
        bind(ILeaderboardProvider.class).to(RedisLeaderboardProvider.class);
        bind(ILeaderboardManager.class).to(SqlLeaderboardManager.class);
    }
}

File: leaderboards/app/modules/LeaderboardsModule.scala
package modules

import com.google.inject.AbstractModule
import interfaces._
import implementors._

case class LeaderboardsModule() extends AbstractModule {
    override def configure() {
        bind(classOf[ILeaderboardProvider]).to(classOf[RedisLeaderboardProvider])
        bind(classOf[ILeaderboardManager]).to(classOf[SqlLeaderboardManager])
    }
}
